src/Hooks.cpp

```diff
@@ -769,6 +771,45 @@ namespace Hooks
 		static inline REL::Relocation<decltype(thunk)> func;
 	};
 
+	struct BSBatchRenderer_RenderPassImmediately1
+	{
+		static void thunk(RE::BSRenderPass* pass, uint32_t technique, bool alphaTest, uint32_t renderFlags)
+		{
+			if (globals::features::lightLimitFix->loaded && !globals::features::lightLimitFix->CheckParticleLights(pass, technique))
+				return;
+
+			func(pass, technique, alphaTest, renderFlags);
+		}
+		static inline REL::Relocation<decltype(thunk)> func;
+	};
+
+	struct BSBatchRenderer_RenderPassImmediately2
+	{
+		static void thunk(RE::BSRenderPass* pass, uint32_t technique, bool alphaTest, uint32_t renderFlags)
+		{
+			if (globals::features::lightLimitFix->loaded && !globals::features::lightLimitFix->CheckParticleLights(pass, technique))
+				return;
+
+			if (globals::features::interiorSunShadows->loaded)
```

Comment:
While I understand the old LLF would only execute func if CheckParticleLights was true. Is it true for your UpdateRasterStateCullMode call? Does this need to be conditioned on particle lights? In other words, would this need to run even if func isn't called?

Reply:
Nah, the engine checks and sets the cull mode including setting dirty flags prior to other render calls so it will automatically be handled. This is just overriding the behaviour for that specific call, and if the call is skipped from failing LLFs checks there's no reason to modify the state.